src/units/abilities.cpp

```diff
@@ -1326,6 +1326,8 @@ bool attack_type::has_weapon_ability(const std::string& special, bool special_id
 			if ( &*it == self_.get() )
 				continue;
 
+			std::vector<special_match> special_tag_matches;
+			std::vector<special_match> special_id_matches;
```

Comment:
I would prefer that these are given different names rather than shadowing an existing name.  That way things like this would be a compiler error instead.

So for example call these `special_tag_matches_adj` and `special_id_matches_adj`, then call the variables on lines 1303-1304 `special_tag_matches_self` and `special_id_matches_self`.

Reply:
done